app/views/forms/odk/_form_controls_node.xml.erb

```diff
@@ -0,0 +1,48 @@
+<% if node.childless? %>
```

Comment:
Yeah I see what you mean. This stuff is tough. A few comments below. Can you also go through and fix the indentation and spacing as it's a bit scattered. That will help readability.